skyportal/models.py

```diff
@@ -733,14 +836,17 @@ def last_detected_at(cls, user):
     @hybrid_method
     def last_detected_mag(self, user):
         """Magnitude at which the object was last detected above a given S/N (3.0 by default)."""
-        detections = [
-            (phot.iso, phot.mag)
-            for phot in Photometry.query_records_accessible_by(user)
+        return (
+            Photometry.query_records_accessible_by(
+                user, columns=[Photometry.mag], mode="read"
+            )
             .filter(Photometry.obj_id == self.id)
-            .all()
-            if phot.snr is not None and phot.snr > PHOT_DETECTION_THRESHOLD
-        ]
-        return max(detections, key=(lambda x: x[0]))[1] if detections else None
+            .filter(Photometry.snr.isnot(None))
+            .filter(Photometry.snr > PHOT_DETECTION_THRESHOLD)
+            .order_by(Photometry.mjd.desc())
+            .limit(1)
+            .scalar()
+        )
```

Comment:
This and the `peak_detected_mag` changes should speed up getting these properties slightly by doing all of the computation on the Postgres side. But I made this change earlier in the optimization process, and now this honestly  doesn't matter too much as we only use these in the single GET method and I stopped using it entirely in the multiple GETs (switched to using the SQL expression versions instead).

Reply:
Would still matter if people are iterating over a list of `obj_id`'s with single GET statements. I actually hope it's not what's happening now...